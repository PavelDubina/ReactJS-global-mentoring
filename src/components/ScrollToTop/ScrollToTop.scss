.scroll_to_top {
  font-size: 48px;
  position: fixed;
  right: 12px;
  bottom: 10px;
  color: #f65261;
  cursor: pointer;
  z-index: 10;
}
