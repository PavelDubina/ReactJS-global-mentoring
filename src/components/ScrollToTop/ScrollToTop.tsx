import React, { useEffect, useState } from 'react'
import { faArrowAltCircleUp } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styles from '../../../styles/ScrollToTop.module.scss'

export const ScrollToTop: React.FC = () => {
  const [isVisible, setIsVisible] = useState(false)

  const toggleVisibility = () => {
    if (window.scrollY > 360) {
      setIsVisible(true)
    } else {
      setIsVisible(false)
    }
  }

  const scrollToTop = () => window.scrollTo({ top: 0, behavior: 'smooth' })

  useEffect(() => {
    window.addEventListener('scroll', toggleVisibility)
    return () => window.removeEventListener('scroll', toggleVisibility)
  })

  return (
    <div className={styles.scroll_to_top}>
      {isVisible && <FontAwesomeIcon data-testid="icon" icon={faArrowAltCircleUp} onClick={scrollToTop} />}
    </div>
  )
}
