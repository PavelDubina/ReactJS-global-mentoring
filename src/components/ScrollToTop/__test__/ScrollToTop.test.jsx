import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { screen } from '@testing-library/dom'
import '@testing-library/jest-dom/extend-expect'
import { ScrollToTop } from '../ScrollToTop'

describe('ScrollToTop', () => {
  const getComponent = () => render(<ScrollToTop />)
  it('Should renders correctly', () => {
    expect(getComponent().asFragment()).toMatchSnapshot()
  })
  it('Should be visible', () => {
    getComponent()
    fireEvent.scroll(window, { target: { scrollY: 361 } })
    expect(screen.queryByTestId('icon')).toBeInTheDocument()
  })
  it('Click on icon should call window.scrollTo', () => {
    const { getByTestId } = getComponent()
    window.scrollTo = jest.fn()
    fireEvent.scroll(window, { target: { scrollY: 361 } })
    fireEvent.click(getByTestId('icon'))
    expect(window.scrollTo).toBeCalled()
  })
  it('Should not be visible', () => {
    getComponent()
    fireEvent.scroll(window, { target: { scrollY: 359 } })
    expect(screen.queryByTestId('icon')).not.toBeInTheDocument()
  })
})
