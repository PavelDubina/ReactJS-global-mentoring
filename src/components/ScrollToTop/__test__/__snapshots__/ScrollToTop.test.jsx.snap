// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="scroll_to_top"
  />
</DocumentFragment>
`;
